(Feature) Search tokens (in Mainnet)

vbaranov opened the pull request:
Relates to https://github.com/poanetwork/metamask-extension/issues/113
Relates to https://github.com/poanetwork/metamask-extension/issues/134

The first step of tokens management support in NW.
The ability to search tokens by name, symbol, contract address. The tokens are from Mainnet only, as for now.

Search screen:

![2018-09-14 17 03 42](https://user-images.githubusercontent.com/4341812/45555975-dbd01680-b842-11e8-94b0-1ed1d68ba778.png)

Confirm screen:

![2018-09-14 17 26 01](https://user-images.githubusercontent.com/4341812/45556187-539e4100-b843-11e8-880d-2d5faafb7725.png)


@dennis00010011b could ypu please update e2e tests to support this feature?

dennis00010011b commented:
@vbaranov 
Add tokens-Search:
 button 'Next' always active if it was once activated
<img width="1440" alt="screen shot 2018-09-14 at 9 30 37 am" src="https://user-images.githubusercontent.com/14067096/45563075-1618c400-b801-11e8-9083-449d77765b31.png">

dennis00010011b commented:
@vbaranov 
Add tokens-Search-Add token:
 app is cropped in full screen mode
<img width="1440" alt="screen shot 2018-09-14 at 9 20 47 am" src="https://user-images.githubusercontent.com/14067096/45563261-a7883600-b801-11e8-9da8-7ea8ef36c4db.png">

dennis00010011b commented:
@vbaranov 
>The tokens are from Mainnet only, as for now.

Mainnet’s tokens is searchable with any provider 
<img width="358" alt="screen shot 2018-09-15 at 3 31 19 am" src="https://user-images.githubusercontent.com/14067096/45585418-afdf8000-b898-11e8-9575-7d85f2888175.png">

dennis00010011b commented:
@vbaranov 
Long tokens name isn't fully displayed  and there is no option to spread it out

<img width="361" alt="screen shot 2018-09-15 at 4 13 13 am" src="https://user-images.githubusercontent.com/14067096/45585721-1fa43980-b89e-11e8-81b8-e9e09212daf7.png">

dennis00010011b commented:
@vbaranov 
Main screen: list of added tokens is not scrollable

<img width="362" alt="screen shot 2018-09-15 at 4 27 01 am" src="https://user-images.githubusercontent.com/14067096/45585801-b6bdc100-b89f-11e8-9af2-d8412a2e6f9c.png">
<img width="361" alt="screen shot 2018-09-15 at 4 25 35 am" src="https://user-images.githubusercontent.com/14067096/45585803-ba514800-b89f-11e8-82a2-4351c117e3c5.png">

dennis00010011b commented:
@vbaranov 
Add token dialog: typo, should be `this token` if one token in the list

<img width="366" alt="screen shot 2018-09-15 at 7 27 45 am" src="https://user-images.githubusercontent.com/14067096/45587322-4328ad80-b8b9-11e8-9d1c-3293a5ab35db.png">

dennis00010011b commented:
@vbaranov 
Add token dialog: it would be helpful to have option to delete some token/tokens from the list before addition

<img width="357" alt="screen shot 2018-09-15 at 7 34 53 am" src="https://user-images.githubusercontent.com/14067096/45587457-95b69980-b8ba-11e8-8ce5-834d67cada17.png">

vbaranov commented:
@dennis00010011b 
>app is cropped in full screen mode

Fixed

>Mainnet’s tokens is searchable with any provider

I've hidden search tab for networks, other than Mainnet

>Long tokens name isn't fully displayed and there is no option to spread it out

Tooltip is added, if a token has long name

>Add token dialog: typo, should be this token if one token in the list

It is not a typo, but now text in the dialog depends on the number of tokens

>Main screen: list of added tokens has no scrollbar
After scrolling down the main menu is invisible

and

>Add token dialog: it would be helpful to have option to delete some token/tokens from the list before addition

Please create feature requests. I suggest to do it in a separate PRs.

dennis00010011b commented:
@vbaranov 
Confirm list is cleared if user goes back from confirm screen to main screen. 
It can be done by clicking  'Cancel' on search screen or close and open wallet.  Is it intentional behavior?

vbaranov commented:
yes, @dennis00010011b this is intentional: pending tokens are cleared on `cancel` or unmounting of view.

In frames of this issue, we also need to fix integration tests. I've added related issue ☝️. Please take a look.

vbaranov commented:
Hi @fvictorio @patitonar - finally, it is ready for review

fvictorio commented:
Can the "searchResults" label be changed to "Search results:"?

![image](https://user-images.githubusercontent.com/417134/45826867-ace6f400-bccb-11e8-85be-5ec1741d2fa4.png)

fvictorio commented:
And maybe "Would you like to add this token?" should be "Would you like to add these tokens?".

patitonar commented on old-ui/app/components/add-token/token-list/token-list.component.js:
```diff
@@ -0,0 +1,72 @@
+                  const title = `${name} (${symbol})`
+                  const isLongTitle = title.length > 28
+
+                  const tokenRow = (key) => h(`.${classnames('token-list__token', {
```
I think we should use JSX here too to be consistent with the rest of the file

vbaranov replied:
Thank you, agree. react-hyperscript was changed to JSX.

vbaranov commented:
>Can the "searchResults" label be changed to "Search results:"?

Just removed the text. It takes some space but has no great meaning.

>And maybe "Would you like to add this token?" should be "Would you like to add these tokens?".

If there are multiple instances of tokens are chosen, you'll see "Would you like to add these tokens?"

>These files seem to have been added by mistake:

Yes, removed them from PR.

Thank you @fvictorio @patitonar for valuable comments. They were processed. I returned back this PR to reviewal stage.
